tutorial-review.md

```diff
@@ -0,0 +1,1053 @@
+```
+
+NB: With a centralized app state we would not need to call the API once again as we would have the post data already available in the client. However, in order not to add extra business logic to pass data on different views or handle refreshing the page, we simply call `/api/posts/post_id` once again.
+
+Here is the page UI:
+
+[react-edit-post]
+
+The only missing action is giving users the ability of creating their own posts. We already created the button "New" in the navbar that redirects to `/posts/new`. 
+Similarly to the previous page `edit.js`, we are prompting a form for the user to fill. Fields are initially empty as we are expecting to store a brand new blog post in the database.
+
+Add a new file `create.js` in `/pages` and copy the following code:
+
+```javascript
+import { useNavigate, Link } from 'react-router-dom';
+import { useForm } from 'react-hook-form';
+import Container from 'react-bootstrap/Container';
+import Button from 'react-bootstrap/Button';
+import Form from 'react-bootstrap/Form';
+import http from '../lib/http';
+
+const Post = () => {
+  const navigate = useNavigate();
+  const { register, handleSubmit } = useForm();
+
+  const onSubmit = async ({ title, author, tags, content }) => {
+    const payload = {
+      title,
+      author,
+      tags: tags.split(',').map((tag) => tag.trim()),
+      content,
+    };
+    await http.post('/api/posts', { data: payload });
+    navigate('/');
+  };
+  
+  return (
+    <Container className="my-5" style={{ maxWidth: '800px' }}>
+      <h1>Create new Post</h1>
+      <Form onSubmit={handleSubmit(onSubmit)} className="my-5">
+        <Form.Group className="mb-3">
+          <Form.Label>Title</Form.Label>
+          <Form.Control type="text" placeholder="Enter title" {...register('title')} />
+        </Form.Group>
+        <Form.Group className="mb-3">
+          <Form.Label>Author</Form.Label>
+          <Form.Control type="text" placeholder="Enter author" {...register('author')} />
+        </Form.Group>
+        <Form.Group className="mb-3">
+          <Form.Label>Tags</Form.Label>
+          <Form.Control type="text" placeholder="Enter tags" {...register('tags')} />
+          <Form.Text className="text-muted">
+            Enter them separately them with ","
+          </Form.Text>
+        </Form.Group>
+        <Form.Group className="mb-3">
+          <Form.Label>Content</Form.Label>
+          <Form.Control as="textarea" rows={3} placeholder="Your content..." {...register('content')} />
+        </Form.Group>
+        <Button variant="primary" type="submit">Publish</Button>
+      </Form>
+      <Link to="/" style={{ textDecoration: 'none' }}>&#8592; Back to Home</Link>
+    </Container>
+  );
+};
+
+export default Post;
+```
+
+The code is very similar to `edit.js`. Here is the UI:
+
+[react-create-post]
+
+Congratulations, we finished the UI as well so we are now ready to deploy our blog app on the internet!
+
+
+## Deploy the app on Koyeb
+
```

Comment:
We should add a step here about creating a GitHub repo and pushing our code to it:

`git init
git remote add origin git@github.com:<YOUR_GITHUB_USERNAME>/<YOUR_GITHUB_REPOSITORY>.git
git add .
git commit -m "Initial commit"
git push -u origin main
`